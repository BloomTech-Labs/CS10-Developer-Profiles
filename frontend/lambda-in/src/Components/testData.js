/**
 * @file Test data
 */

/**
 * User test data.
 * @const
 * @type {object}
 */
export const DEV_TEST_DATA = {
  _id: '5ba390842afca2160cb0ffbe',
  __v: 0,
  firstName: 'Jean Surkouf',
  lastName: 'Ariza Varela',
  summary: 'CS and Software development passionate.',
  email: 'reg@c14.com',
  password: '$2b$12$r/CTYDAaLzq7VtWMwaFNdOJhPYf/9/Py25irLgUfcZWsti2xHLgN6',
  desiredTitle: 'Full Stack Software Developer',
  topSkills: ['React', 'TDD', 'LESS', 'CS', 'HTML', 'HAPPY_CODING'],
  additionalSkills: ['Passionate', 'Happy'],
  familiarWith: ['Python', 'C', 'Djangl'],
  github: 'https://github.com/talent_developer',
  linkedin: 'Linkedin/talent_developer',
  portfolio: 'portfolio.talent_developer',
  placesInterested: [
    {
      loc: {
        coordinates: ['65.98', '-55.666'],
        type: 'Point',
      },
      _id: '5ba39769cc47111861ac5574',
      place: 'Califirnia',
    },
    {
      loc: {
        coordinates: ['-0.666', '-66.666'],
        type: 'Point',
      },
      _id: '5ba39769cc47111861ac5573',
      place: 'Canada',
    },
  ],

  projects: [
    {
      tech: ['NODE', 'Express', 'Mongoose', 'React'],
      _id: '5bbcc2aee991c12f8e4bde3a',
      title: 'Dev profiles',
      description: 'A meeting place for employers and talent developers',
      img: '/nice.img.edit',
      link: 'go.to.project.com',
      repo: 'github.my-repo',
    },
    {
      tech: ['JavaScript', 'Express', 'Mongoose', 'NPM'],
      _id: '5bbcc2aee991c12f8e4bde39',
      title: 'Router Factroy',
      description: 'A DRY library for express-mongoose',
      img: '/nice.img',
      link: 'go.to.project.com',
      repo: 'github.my-repo',
    },
  ],
  experience: [
    {
      _id: '5bbcc2aee991c12f8e4bde3d',
      title: 'LCAI',
      company: 'UAM research group',
      location: {
        loc: {
          coordinates: ['-0.666', '-66.666'],
          type: 'Point',
        },
        _id: '5bb748545cdfe308de503a2b',
        place: 'Madrid',
      },
      headline: 'Wedsite for an research group.',
      description:
        'A dynamic website with no know CRM and Backend hosted in google Spreadsheet.',
      startYear: '2018-09-27T19:10:21.809Z',
      endYear: '2018-09-27T19:10:21.809Z',
    },
    {
      _id: '5bbcc2aee991c12f8e4bde3b',
      title: 'Developer profiles',
      company: 'Lambda',
      location: {
        loc: {
          coordinates: ['10.666', '43.666'],
          type: 'Point',
        },
        _id: '5bb748545cdfe308de503a29',
        place: 'Califormia',
      },
      headline: 'Meeting point for employers looking for talent developers.',
      description: 'A place to match employer needs with a talent developer',
      startYear: '2018-09-27T19:10:21.809Z',
      endYear: '2018-10-12T19:10:21.809Z',
    },
  ],
  education: [
    {
      _id: '5bbcc2aee991c12f8e4bde40',
      school: 'Lambda School',
      degree: 'Software developer',
      fieldOfStudy: 'CS',
      startYear: '2018-03-31T22:00:00.000Z',
      endYear: '2018-10-11T22:00:00.000Z',
      description:
        'A CS and Software development master with a focus on web technologies.',
    },
    {
      _id: '5bbcc2aee991c12f8e4bde3f',
      school: 'UNED',
      degree: 'Economics',
      fieldOfStudy: 'Social sciences',
      startYear: '2012-03-31T22:00:00.000Z',
      endYear: '2017-10-11T22:00:00.000Z',
      description: 'Economics studies.',
    },
  ],
};

export const EMPLOYER_TEST_DATA = {};
