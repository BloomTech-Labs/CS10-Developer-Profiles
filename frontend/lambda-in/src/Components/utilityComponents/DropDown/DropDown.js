import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Icon from '@material-ui/core/Icon';

/**
 * A wrapper component for @material-ui <ExpansionPanel />
 *
 * @description A Dropdown component wrapper, which accepts any type of children.
 *
 * @function A functional component.
 * @prop {string} header - Text to be displayed when the Dropdown is collapsed.
 *
 * @example How to use it
 * <DropDown header="Header text to display here">
 */
const DropDown = ({ header, children }) => (
  <Fragment>
    <div className="panelMargin">
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<Icon>add_circle</Icon>}>
          <Typography>{header}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div className="expansionPanel">{children}</div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </div>
  </Fragment>
);

DropDown.propTypes = {
  header: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]).isRequired,
};

export default DropDown;
