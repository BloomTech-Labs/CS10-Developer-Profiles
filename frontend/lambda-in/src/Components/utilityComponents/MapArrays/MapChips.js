import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import DeleteForever from '@material-ui/icons/DeleteForever';
import Chip from '@material-ui/core/Chip';

/**
 * Render a list of @material-ui's Chip components.
 *
 * @function
 * @callback removeItem
 * @prop {String[]|Number[]} props - Array of Strings.
 * @prop {object} props.classes - Object defining the classes needed.
 * @prop {object} props.classes.container - Class for the 'container' Node.
 * @prop {object} props.classes.item - Class for each 'item' Node.
 * @prop {removeItem} props.removeItem - Handler to delete and item.
 * @prop {string} props.field - The user property to handle.
 * @return {object} A React component
 */

const MapChips = (props) => {
  // prettier-ignore
  const {
    field,
    classes,
    array,
    removeItem,
  } = props;

  return (
    <Grid
      container
      spacing={16}
      className={classes.container}
      alignItems="center"
      direction="row"
      justify="center"
    >
      {array.map((value, index) => (
        <div key={`${value}${Math.random()}`}>
          <Chip
            id={`${index}$${value}`}
            label={`${value}`}
            className={classes.item}
            deleteIcon={<DeleteForever />}
            onDelete={removeItem && removeItem(field, index)}
          />
        </div>
      ))}
    </Grid>
  );
};

MapChips.propTypes = {
  array: PropTypes.arrayOf(PropTypes.any).isRequired,
  classes: PropTypes.objectOf(PropTypes.string),
  removeItem: PropTypes.func,
  field: PropTypes.string,
};
MapChips.defaultProps = {
  classes: {},
  removeItem: null,
  field: null,
};

export default MapChips;
