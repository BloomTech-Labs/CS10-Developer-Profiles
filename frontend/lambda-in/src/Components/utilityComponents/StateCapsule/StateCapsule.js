import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Create a local state to manage an object's properties whilist is edited.
 *
 * @description
 */
class StateCapsule extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ready: false,
    };
    this.removeItem = this.removeItem.bind(this);
    this.createItem = this.createItem.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    // prettier-ignore
    this.mapAndResetKeysValues = object => Object.keys(object)
      .reduce((resetKeys, key) => {
        const valueType = Object.prototype.toString.call(object[key]);

        switch (valueType) {
          case '[object Array]':
            // eslint-disable-next-line no-param-reassign
            resetKeys[key] = [];
            // eslint-disable-next-line no-param-reassign
            resetKeys[`${key}_edit`] = '';
            break;
          default:
            // eslint-disable-next-line no-param-reassign
            resetKeys[key] = '';
        }

        return resetKeys;
      }, {});
  }

  componentDidMount() {
    this.resetState();
  }

  resetState() {
    const { schema, object } = this.props;
    const newState = {
      ...this.mapAndResetKeysValues(schema),
      ...object,
      ready: true,
    };

    // Set local state with object-properties and initialize its values.
    this.setState(newState);
  }

  /**
   * Sync local state with input field.
   */
  handleChange(event) {
    event.stopPropagation();

    const { id } = event.target;
    const details = id.split('-');
    const field = details[1];
    const { value } = event.target;

    this.setState({ [field]: value });
  }

  handleKeyPress(e) {
    e.stopPropagation();
    const { chips } = e.target.dataset;

    if (e.key === 'Enter' && chips) {
      e.preventDefault();

      const details = e.target.id.split('-');
      const { field } = e.target.dataset;
      const { value } = e.target;

      this.setState((prevState) => {
        const toUpdate = prevState[field];
        toUpdate.push(value);
        return { [field]: toUpdate, [details[1]]: '' };
      });
    }
  }

  createItem(field) {
    const newData = { ...this.state };
    delete newData.ready;

    const createEvent = new CustomEvent('onCreateItem', {
      bubbles: true,
      detail: {
        newData,
        field,
      },
    });
    return (e) => {
      e.stopPropagation();

      e.target.dispatchEvent(createEvent);
      this.resetState();
    };
  }

  removeItem(field, index) {
    const removeEvent = new CustomEvent('onDeleteItem', {
      bubbles: true,
      detail: { field, index },
    });

    return (e) => {
      e.stopPropagation();

      //
      e.target.dispatchEvent(removeEvent);

      const typeOfField = Object.prototype.toString.call(field);

      if (
        // prettier-ignore
        typeOfField === '[object String]'
        || typeOfField === '[object Number]'
        || typeOfField === '[object Boolean]'
      ) {
        this.setState((prevState) => {
          const toUpdate = prevState[field];
          toUpdate.splice(index, 1);
          return { [field]: toUpdate };
        });
      }
    };
  }

  render() {
    const { ready } = this.state;

    const stateCapsule = { ...this.state };
    delete stateCapsule.ready;

    const { children } = this.props;

    return ready ? (
      // eslint-disable-next-line jsx-a11y/no-static-element-interactions
      <div
        className="state-capsule"
        onChange={this.handleChange}
        onKeyPress={this.handleKeyPress}
      >
        {children({
          stateCapsule,
          removeItem: this.removeItem,
          createItem: this.createItem,
        })}
      </div>
    ) : null;
  }
}

StateCapsule.propTypes = {
  schema: PropTypes.shape({}).isRequired,
  object: PropTypes.shape({}).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]).isRequired,
};

export default StateCapsule;
