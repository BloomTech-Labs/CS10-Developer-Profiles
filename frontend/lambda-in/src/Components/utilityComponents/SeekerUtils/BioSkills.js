import React, { Fragment } from 'react';
import DropDown from '../DropDown';
import TextField from '@material-ui/core/TextField';

const BioSkills = props => {
  const { userInfo } = props;

  return (
    <Fragment>
      <DropDown header="tell us something about yourself!">
        <div className="inputFieldLargeMultiline">
          <TextField
            id="summary"
            label="Your Bio"
            fullWidth
            multiline
            rowsMax="4"
            value={userInfo.summary}
            margin="normal"
            variant="outlined"
          />
        </div>
        <div className="inputFieldLargeMultiline">
          <TextField
            id="topSkills"
            label="Top skills"
            fullWidth
            multiline
            rowsMax="4"
            value={userInfo.topskill}
            margin="normal"
            variant="outlined"
          />
        </div>
      </DropDown>
    </Fragment>
  );
};

export default BioSkills;
