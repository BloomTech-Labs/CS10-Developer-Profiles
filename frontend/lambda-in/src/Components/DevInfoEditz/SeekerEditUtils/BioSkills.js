import TextField from '@material-ui/core/TextField';
import React from 'react';
import SEEKER_PROP_TYPES from '../../seekerPropTypes';
import DropDown from '../../utilityComponents/DropDown/DropDown';
import MapChips from '../../utilityComponents/MapArrays/MapChips';
import StateCapsule from '../../utilityComponents/StateCapsule/StateCapsule';

/**
 * Render the Developer's 'summary' and 'topSkills' fields.
 *
 * @function
 * @return {object} A React component.
 */
const BioSkills = ({ userInfo }) => {
  const field = 'topSkills';
  const schema = {
    summary: userInfo.summary,
    topSkills: userInfo.topSkills,
  };

  return (
    <StateCapsule schema={schema} object={schema}>
      {({ stateCapsule, removeItem }) => (
        <DropDown header="Biio and Skills">
          <div className="inputFieldLargeMultiline">
            <TextField
              id="edit-summary"
              label="Your Bio"
              fullWidth
              multiline
              rowsMax="20"
              value={stateCapsule.summary}
              margin="normal"
              variant="outlined"
            />
          </div>
          <div className="inputFieldLargeMultiline">
            <TextField
              id="edit-topSkills_edit"
              label="Add skills - press enter"
              fullWidth
              multiline
              rowsMax="4"
              value={stateCapsule.topSkills_edit}
              margin="normal"
              variant="outlined"
              inputProps={{
                // info for the DevInfoEditz's onBlur handler.
                'data-chips': true,
                'data-field': field,
                'data-itemType': 'string',
                'data-value': stateCapsule.topSkills.join('-'),
              }}
            />
          </div>
          {/* <MapUnorderedList array={arr} /> */}
          <MapChips
            field={field}
            array={stateCapsule.topSkills}
            removeItem={removeItem}
          />
          {/* TODO: Add summary - enable uploading a file */}
        </DropDown>
      )}
    </StateCapsule>
  );
};

BioSkills.propTypes = {
  userInfo: SEEKER_PROP_TYPES.isRequired,
};

export default BioSkills;
