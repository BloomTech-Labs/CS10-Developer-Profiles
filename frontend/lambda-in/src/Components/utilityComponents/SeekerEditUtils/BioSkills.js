import React, { Fragment } from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import DropDown from '../DropDown/DropDown';
import MapChips from '../MapArrays/MapChips';
import ArrayController from '../MapArrays/ArrayController';
import StateCapsule from '../StateCapsule/StateCapsule';

/**
 * Render the Developer's 'summary' and 'topSkills' fields.Fragment
 *
 * @function
 * @return {object} A React component.
 */
// eslint-disable-next-line arrow-parens
const BioSkills = props => {
  // eslint-disable-next-line react/prop-types
  const { userInfo, setFS } = props;

  const field = 'topSkills';
  const schema = {
    summary: userInfo.summary,
    topSkills: userInfo.topSkills,
  };

  return (
    <StateCapsule schema={schema} object={schema}>
      {({ stateCapsule }) => (
        <DropDown header="Biio and Skills">
          <div className="inputFieldLargeMultiline">
            <TextField
              id="new-summary"
              index="an index"
              label="Your Bio"
              fullWidth
              multiline
              rowsMax="20"
              value={stateCapsule.summary}
              margin="normal"
              variant="outlined"
            />
          </div>
          {/* TODO */}
          <ArrayController
            itemSchema="singleItem"
            title="Top Skills"
            field={field}
            setPFS={setFS}
            arr={stateCapsule.topSkills}
          >
            {/* eslint-disable-next-line object-curly-newline */}
            {({
              handleChange,
              updateFormState,
              removeItem,
              state,
              arr,
              title,
            }) => (
              <Fragment>
                <div
                  className="inputFieldLargeMultiline"
                  onChange={handleChange}
                >
                  <TextField
                    id="newItem"
                    label="Add skills"
                    fullWidth
                    multiline
                    rowsMax="4"
                    value={state.newItem}
                    margin="normal"
                    variant="outlined"
                    inputProps={{
                      'data-value': state.newItem,
                      'data-field': field,
                    }}
                  />
                </div>
                {/* <MapUnorderedList array={arr} /> */}
                <MapChips array={arr} removeItem={removeItem} />
                <Button
                  variant="outlined"
                  color="primary"
                  align="center"
                  onClick={updateFormState}
                >
                  {`Add new ${title}`}
                </Button>
              </Fragment>
            )}
          </ArrayController>

          {/* TODO: Add summary - enable uploading a file */}
        </DropDown>
      )}
    </StateCapsule>
  );
};

export default BioSkills;
