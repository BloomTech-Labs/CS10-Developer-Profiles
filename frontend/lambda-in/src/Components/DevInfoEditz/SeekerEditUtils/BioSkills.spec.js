import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BioSkills from './BioSkills';
import TestWrapper from '../../utilityComponents/TestWrapper/TestWrapper';
import StateCapsule from '../../utilityComponents/StateCapsule/StateCapsule';
import DropDown from '../../utilityComponents/DropDown/DropDown';
import MapChips from '../../utilityComponents/MapArrays/MapChips';
import { DEV_TEST_DATA } from '../../testData';

configure({ adapter: new Adapter() });

/**
 * @author jesuarva
 * We are only testing proper rendering of inner components,
 * this is accomplish with a snapshot and some specific test.
 *
 * @see StateCapsule.spec.js for specific test related to its fucntionality.
 * @see DropDown.spec.js for specific test related to its fucntionality.
 * @see MapChips.spec.js for specific test related to its fucntionality.
 */
describe('BioSkills component', () => {
  const bioSkills = <BioSkills userInfo={DEV_TEST_DATA} />;

  it('should renders correctly', () => {
    const tree = renderer.create(bioSkills);
    expect(tree).toMatchSnapshot();
  });

  describe('Renders children components', () => {
    const mounted = mount(<TestWrapper>{() => bioSkills}</TestWrapper>);

    it('should renders a SteceCapsule component', () => {
      const stateCapsule = mounted.find(StateCapsule);
      expect(stateCapsule).toHaveLength(1);
    });

    it('should renders a DropDown component', () => {
      const dropDown = mounted.find(DropDown);
      expect(dropDown).toHaveLength(1);
    });

    it('should renders a MapChips component', () => {
      const mapChips = mounted.find(MapChips);
      expect(mapChips).toHaveLength(1);
    });
  });
});
