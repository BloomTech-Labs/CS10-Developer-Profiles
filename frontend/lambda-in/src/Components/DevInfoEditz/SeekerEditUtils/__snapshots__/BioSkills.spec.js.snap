// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioSkills component should renders correctly 1`] = `
<div
  className="state-capsule"
  onChange={[Function]}
  onKeyPress={[Function]}
>
  <div
    className="panelMargin"
  >
    <div
      className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
    >
      <div
        aria-expanded={false}
        className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiExpansionPanelSummary-content-35"
        >
          <p
            className="MuiTypography-root-40 MuiTypography-body1-49"
          >
            Biio and Skills
          </p>
        </div>
        <div
          aria-hidden="true"
          className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label-71"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-72"
            >
              add_circle
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-81"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        className="MuiCollapse-container-88"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper-90"
        >
          <div
            className="MuiCollapse-wrapperInner-91"
          >
            <div
              className="MuiExpansionPanelDetails-root-92"
            >
              <div
                className="expansionPanel"
              >
                <div
                  className="inputFieldLargeMultiline"
                >
                  <div
                    aria-describedby={undefined}
                    className="MuiFormControl-root-93 MuiFormControl-marginNormal-94 MuiFormControl-fullWidth-96"
                  >
                    <label
                      className="MuiFormLabel-root-104 MuiFormLabel-filled-108 MuiInputLabel-root-97 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-shrink-100 MuiInputLabel-outlined-103"
                      data-shrink={true}
                      htmlFor="edit-summary"
                    >
                      Your Bio
                    </label>
                    <div
                      className="MuiInputBase-root-124 MuiOutlinedInput-root-111 MuiInputBase-fullWidth-133 MuiInputBase-formControl-125 MuiInputBase-multiline-132 MuiOutlinedInput-multiline-117"
                      onClick={[Function]}
                    >
                      <fieldset
                        aria-hidden={true}
                        className="MuiNotchedOutline-root-141 MuiOutlinedInput-notchedOutline-118"
                        style={
                          Object {
                            "paddingLeft": 8,
                          }
                        }
                      >
                        <legend
                          className="MuiNotchedOutline-legend-142"
                          style={
                            Object {
                              "width": NaN,
                            }
                          }
                        />
                      </fieldset>
                      <div
                        className="Textarea-root-146"
                      >
                        <textarea
                          aria-hidden="true"
                          className="Textarea-textarea-147 Textarea-shadow-148"
                          readOnly={true}
                          rows="1"
                          tabIndex={-1}
                          value=""
                        />
                        <textarea
                          aria-hidden="true"
                          className="Textarea-textarea-147 Textarea-shadow-148"
                          defaultValue={undefined}
                          readOnly={true}
                          rows={1}
                          tabIndex={-1}
                          value="CS and Software development passionate."
                        />
                        <textarea
                          aria-invalid={false}
                          autoComplete={undefined}
                          autoFocus={undefined}
                          className="Textarea-textarea-147 MuiInputBase-input-134 MuiOutlinedInput-input-119 MuiInputBase-inputMultiline-136 MuiOutlinedInput-inputMultiline-121"
                          defaultValue={undefined}
                          disabled={false}
                          id="edit-summary"
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={undefined}
                          onKeyUp={undefined}
                          placeholder={undefined}
                          readOnly={undefined}
                          required={false}
                          rows={1}
                          style={
                            Object {
                              "height": 19,
                            }
                          }
                          value="CS and Software development passionate."
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="inputFieldLargeMultiline"
                >
                  <div
                    aria-describedby={undefined}
                    className="MuiFormControl-root-93 MuiFormControl-marginNormal-94 MuiFormControl-fullWidth-96"
                  >
                    <label
                      className="MuiFormLabel-root-104 MuiInputLabel-root-97 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-outlined-103"
                      data-shrink={false}
                      htmlFor="edit-topSkills_edit"
                    >
                      Add skills - press enter
                    </label>
                    <div
                      className="MuiInputBase-root-124 MuiOutlinedInput-root-111 MuiInputBase-fullWidth-133 MuiInputBase-formControl-125 MuiInputBase-multiline-132 MuiOutlinedInput-multiline-117"
                      onClick={[Function]}
                    >
                      <fieldset
                        aria-hidden={true}
                        className="MuiNotchedOutline-root-141 MuiOutlinedInput-notchedOutline-118"
                        style={
                          Object {
                            "paddingLeft": NaN,
                          }
                        }
                      >
                        <legend
                          className="MuiNotchedOutline-legend-142"
                          style={
                            Object {
                              "width": 0.01,
                            }
                          }
                        />
                      </fieldset>
                      <div
                        className="Textarea-root-146"
                      >
                        <textarea
                          aria-hidden="true"
                          className="Textarea-textarea-147 Textarea-shadow-148"
                          readOnly={true}
                          rows="1"
                          tabIndex={-1}
                          value=""
                        />
                        <textarea
                          aria-hidden="true"
                          className="Textarea-textarea-147 Textarea-shadow-148"
                          defaultValue={undefined}
                          readOnly={true}
                          rows={1}
                          tabIndex={-1}
                          value=""
                        />
                        <textarea
                          aria-invalid={false}
                          autoComplete={undefined}
                          autoFocus={undefined}
                          className="Textarea-textarea-147 MuiInputBase-input-134 MuiOutlinedInput-input-119 MuiInputBase-inputMultiline-136 MuiOutlinedInput-inputMultiline-121"
                          data-chips={true}
                          data-field="topSkills"
                          data-itemtype="string"
                          data-value="React-TDD-LESS-CS-HTML-HAPPY_CODING"
                          defaultValue={undefined}
                          disabled={false}
                          id="edit-topSkills_edit"
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={undefined}
                          onKeyUp={undefined}
                          placeholder={undefined}
                          readOnly={undefined}
                          required={false}
                          rows={1}
                          style={
                            Object {
                              "height": 19,
                            }
                          }
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-container-149 MuiGrid-spacing-xs-16-172 MuiGrid-align-items-xs-center-157 MuiGrid-justify-xs-center-166"
                >
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="0$React"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        React
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="1$TDD"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        TDD
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="2$LESS"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        LESS
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="3$CS"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        CS
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="4$HTML"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        HTML
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                  <div>
                    <div
                      className="MuiChip-root-246 MuiChip-deletable-252"
                      id="5$HAPPY_CODING"
                      onClick={undefined}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="MuiChip-label-265"
                      >
                        HAPPY_CODING
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-271 MuiChip-deleteIcon-266"
                        color={undefined}
                        focusable="false"
                        onClick={[Function]}
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
